plugins/outputs/sql/sql.go

```diff
@@ -294,3 +311,48 @@ func newSQL() *SQL {
 		ConnectionMaxIdle: 2,
 	}
 }
+
+// Convert a DSN possibly using v1 parameters to clickhouse-go v2 format
+func convertClickHouseDsn(dsn string, log telegraf.Logger) string {
+	if p, err := url.Parse(dsn); err == nil {
```

Comment:
How about inverting the logic here to save an indentation level...
```suggestion
	p, err := url.Parse(dsn)
	if err != nil {
		return dsn
	}
```